src/librustc_trans/back/link.rs

```diff
@@ -938,8 +938,10 @@ fn add_upstream_rust_crates(cmd: &mut Linker,
             Linkage::NotLinked |
             Linkage::IncludedFromDylib => {}
             Linkage::Static => {
+                let is_a_no_builtins_crate =
+                    attr::contains_name(&sess.cstore.crate_attrs(cnum), "no_builtins");
```

Comment:
So that struct is not really usable because in all these contexts the only thing we know about the dependency is its `CrateNum` (basically an ID number) and there's no way to get a `CrateTranslation` from it. Even if that were possible it would be more expensive, I think, that the current `attr::contains_name` call. However, an improvement we could do is to change this `attr::contains_name` call for something more readable like `sess.cstore.is_no_builtins_crate(cnum)` everywhere.